File: MultiLanguageSandbox/src/test/deps/java/25.java

class Solution {
  /**
  * Given an integer n, return the minimum number of operations required to make all elements in the array arr equal.
  * The array arr has a length of n, and arr[i] = (2 * i) + 1 for 0 <= i < n.
  * In one operation, you can select two indices x and y (0 <= x, y < n) and subtract 1 from arr[x] and add 1 to arr[y] (i.e., arr[x] -= 1 and arr[y] += 1).
  * The goal is to make all elements in arr equal. The test cases will ensure that after performing some operations, all elements in arr can be made equal.
  *
  * Example 1:
  * Input: n = 3
  * Output: 2
  * Explanation: arr = [1, 3, 5]
  * First operation: x = 2, y = 0 -> arr = [2, 3, 4]
  * Second operation: x = 2, y = 0 -> arr = [3, 3, 3]
  *
  * Example 2:
  * Input: n = 6
  * Output: 9
  */
  public int minOperations(int n) 
{
    int operations = 0;
    int target = (2 * (n / 2)) + 1; // For odd n, this is the middle element; for even, it's the higher middle
    if (n % 2 == 0) {
        target = (2 * (n / 2 - 1)) + 1 + 1; // Adjust for even n: average of two middle elements is (middle1 + middle2)/2, but since operations must result in integers, the target is the higher middle for even n (e.g., for n=6, middle elements are 5 and 7, target is 6)
        // Wait, no. For n=6, the array is [1,3,5,7,9,11]. The target is (5+7)/2 =6. But operations would be sum of (6-1)+(6-3)+(6-5) =5+3+1=9. So the target is indeed 6.
        // So for even n, the target is (2*(n/2 -1) +1 + 2*(n/2) +1)/2 = ( (2k-1) + (2k+1) )/2 = (4k)/2 = 2k, where k is n/2. So for n=6, k=3, 2*3=6.
        target = 2 * (n / 2);
    } else {
        target = (2 * (n / 2)) + 1;
    }
    
    for (int i = 0; i < n / 2; i++) {
        int current = (2 * i) + 1;
        operations += (target - current);
    }
    return operations;
}
  public static void main(String[] args) {
      Solution solution = new Solution();

      // Test the function with different test cases
      assert solution.minOperations(4) == 4;
      assert solution.minOperations(5) == 6;
      assert solution.minOperations(7) == 12;
      System.out.println("All tests passed");
  }
}

File: MultiLanguageSandbox/src/test/deps/dart/25.dart

int countRemainingTrees(int roadLength, List<List<int>> regions)
/*Problem Description
Adjacent to a school's main gate, there is a road of length $l$ meters, lined with a sequence of trees. The spacing between each pair of adjacent trees is 1 meter. The road can be conceptualized as a numerical axis, with one end at the position 0 on the axis and the other end at position $l$; a tree is planted at every integer point on this axis, i.e., at $0,1,2,\dots,l$.
Some sections of this road are designated for subway construction. These sections are defined by their starting and ending points on the numerical axis. It is known that the coordinates of any section's starting and ending points are integers, and there may be overlaps between sections. The current task involves removing the trees within these sections, including the trees located at the endpoints of each section. Your objective is to calculate the number of trees remaining on the road after these trees have been removed.

Input Format
The first line contains two integers, representing respectively the length of the road $l$ and the number of sections $m$.
The following $m$ lines each contain two integers $u, v$, representing the coordinates of a section's starting and ending points.

Output Format
Output a single integer on one line, representing the number of trees remaining on the road after the specified trees have been removed.

Sample 
countRemainingTrees(500,[[150, 300], [301, 450]]) => 200
*/
{
  Set<int> removedPositions = Set();
  
  for (var region in regions) {
    int u = region[0];
    int v = region[1];
    // Ensure u <= v to handle cases where u might be greater than v
    if (u > v) {
      int temp = u;
      u = v;
      v = temp;
    }
    // Add all positions from u to v inclusive to the set
    for (int i = u; i <= v; i++) {
      removedPositions.add(i);
    }
  }
  
  int totalTrees = roadLength + 1;
  int removedCount = removedPositions.length;
  return totalTrees - removedCount;
}
void testCountRemainingTrees() {
  // Test case 1: Non-overlapping regions
  int roadLength1 = 500;
  List<List<int>> regions1 = [
    [150, 300],
    [301, 450]
  ];
  int expectedRemaining1 = 200; // Trees from 0-149, 451-500
  assert(countRemainingTrees(roadLength1, regions1) == expectedRemaining1);

  // Test case 2: Partially overlapping regions
  int roadLength2 = 500;
  List<List<int>> regions2 = [
    [100, 200],
    [150, 300],
    [470, 471]
  ];
  int expectedRemaining2 = 298; // Trees from 0-99, 301-469, 472-500
  assert(countRemainingTrees(roadLength2, regions2) == expectedRemaining2);

  // Test case 3: All trees are removed
  int roadLength3 = 10;
  List<List<int>> regions3 = [
    [0, 10]
  ];
  int expectedRemaining3 = 0; // All trees are removed
  assert(countRemainingTrees(roadLength3, regions3) == expectedRemaining3);

  print("All test cases passed successfully.");
}

void main() {
  testCountRemainingTrees();
}